zerver/lib/streams.py

```diff
@@ -411,10 +410,13 @@ def check_stream_access_for_delete_or_update(
     if user_profile.is_realm_admin:
         return
 
+    if can_administer_channel(stream, user_profile, sub):
+        return
+
     if sub is None and stream.invite_only:
         raise JsonableError(error)
 
-    raise OrganizationAdministratorRequiredError
+    raise JsonableError(_("Insufficient permission"))
```

Comment:
I think we probably want something more specific. At the very least, let's define a coded subclass in `zerver/lib/exceptions.py`, maybe `CANNOT_ADMINISTER_CHANNEL/CannotAdministerChannelError`, so that clients that might want to understand this exception can do so.

"You do not have permission to administer this channel." is probably a fine string to go with for now.

But future code is going to get more complicated because we'll be checking both this and whether you have some more specific permission on the channel.